internal/storage/v2/elasticsearch/tracestore/from_dbmodel.go

```diff
@@ -25,22 +25,70 @@ import (
 var errType = errors.New("invalid type")
 
 // FromDBModel converts multiple ES DB Spans to internal traces
-func FromDBModel(spans []dbmodel.Span) (ptrace.Traces, error) {
+func FromDBModel(spans []dbmodel.Span, tagDotReplacement string) (ptrace.Traces, error) {
 	traceData := ptrace.NewTraces()
 	if len(spans) == 0 {
 		return traceData, nil
 	}
-
+	fromDb := &fromDBModel{tagDotReplacement: tagDotReplacement}
 	resourceSpans := traceData.ResourceSpans()
 	resourceSpans.EnsureCapacity(len(spans))
-	err := dbSpansToSpans(spans, resourceSpans)
+	err := fromDb.dbSpansToSpans(spans, resourceSpans)
 
 	return traceData, err
 }
 
-func dbProcessToResource(process dbmodel.Process, resource pcommon.Resource) {
+type fromDBModel struct {
+	tagDotReplacement string
+}
+
+func (f *fromDBModel) convertTagField(k string, v any) dbmodel.KeyValue {
```

Comment:
utility function should not be at the top of the file, it breaks the flow of logic